tests/compas/test_units_geometry.py

```diff
@@ -0,0 +1,314 @@
+        point_data = {
+            'geometry': p,
+            'units': 'meter',
+            'precision': 0.001 * units.meter
+        }
+        
+        # Should serialize correctly
+        json_str = json.dumps(point_data, cls=DataEncoder)
+        reconstructed = json.loads(json_str, cls=DataDecoder)
+        
+        assert isinstance(reconstructed['geometry'], Point)
+        assert reconstructed['units'] == 'meter'
+        assert hasattr(reconstructed['precision'], 'magnitude')
+        assert reconstructed['precision'].magnitude == 0.001
+    
+    def test_vector_workflow_with_units(self):
+        """Test Vector in unit-aware workflows."""
+        # Vectors are created with plain coordinates
+        v = Vector(1.0, 2.0, 3.0)
+        
+        # But can be part of unit-aware data
+        vector_data = {
+            'direction': v,
+            'magnitude': 5.0 * units.meter,
+            'force': 100.0 * units.Quantity(1.0, 'newton')
+        }
+        
+        # Should serialize correctly
+        json_str = json.dumps(vector_data, cls=DataEncoder)
+        reconstructed = json.loads(json_str, cls=DataDecoder)
+        
+        assert isinstance(reconstructed['direction'], Vector)
+        assert hasattr(reconstructed['magnitude'], 'magnitude')
+        assert reconstructed['magnitude'].magnitude == 5.0
+    
+    def test_frame_with_unit_context(self):
+        """Test Frame in unit-aware context."""
+        frame = Frame([1.0, 2.0, 3.0])
+        
+        # Frame can be part of unit-aware design data
+        design_data = {
+            'coordinate_frame': frame,
+            'scale': 1.0 * units.meter,
+            'tolerance': 0.01 * units.meter
+        }
+        
+        # Should serialize correctly
+        json_str = json.dumps(design_data, cls=DataEncoder)
+        reconstructed = json.loads(json_str, cls=DataDecoder)
+        
+        assert isinstance(reconstructed['coordinate_frame'], Frame)
+        assert hasattr(reconstructed['scale'], 'magnitude')
+        assert reconstructed['scale'].magnitude == 1.0
+
+
+@pytest.mark.skipif(not UNITS_AVAILABLE, reason="pint not available")
+class TestMeshWithUnitsWorkflow:
+    """Test Mesh in unit-aware workflows."""
+    
+    def test_mesh_with_unit_metadata(self):
+        """Test Mesh with unit metadata."""
+        # Create simple mesh
+        vertices = [[0.0, 0.0, 0.0], [1.0, 0.0, 0.0], [0.0, 1.0, 0.0]]
+        faces = [[0, 1, 2]]
+        mesh = Mesh.from_vertices_and_faces(vertices, faces)
+        
+        # Add unit-aware metadata
+        mesh_data = {
+            'mesh': mesh,
+            'units': 'meter',
+            'scale_factor': 1.0 * units.meter,
+            'material_thickness': 0.1 * units.meter,
+            'area': 0.5 * units.Quantity(1.0, 'meter^2')
+        }
+        
+        # Should serialize correctly
+        json_str = json.dumps(mesh_data, cls=DataEncoder)
+        reconstructed = json.loads(json_str, cls=DataDecoder)
```

Comment:
Instead of this custom dict of mesh_data, use the Mesh's custom attributes to add custom data